### PR TITLE
[TOOLS-2321] Taskboard now updates new projects when updating the cache

### PR DESCRIPTION
**Pull request by emannuelalmeida**

### CODE REVIEW

File: core/src/main/java/objective/taskboard/issueBuffer/IssueBufferService.java [hunk 1]
```diff
@@ -157,7 +160,21 @@ public synchronized void updateIssueBuffer() {
         }
     }
 
-    private void updateState(IssueBufferState state) {
+    private void checkAndUpdateNewProjectsIssues(IssueBufferServiceSearchVisitor visitor) {
+        if (cardsRepo.getCurrentProjects().isPresent()) {
+            List<ProjectFilterConfiguration> projects = projectService.getTaskboardProjects();
+            Set<String> projectsOnCache = cardsRepo.getCurrentProjects().get();
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/issueBuffer/IssueBufferService.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3655)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: core/src/main/java/objective/taskboard/issueBuffer/CardRepo.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
     private Set<String> unsavedCards = new HashSet<>();
     private Set<String> currentProjects = new HashSet<>();
     private Map<Long, String> keyById = new ConcurrentHashMap<>();
-       
+
```
Comment (jhonypg) [inline core/src/main/java/objective/taskboard/issueBuffer/CardRepo.java]: Remover modificação.

File: core/src/main/java/objective/taskboard/issueBuffer/IssueBufferService.java [hunk 3]
```diff
@@ -157,7 +159,21 @@ public synchronized void updateIssueBuffer() {
...
+            List<String> newProjectKeys = projectService.getTaskboardProjects().stream()
+                    .map(pr -> pr.getProjectKey())
+                    .filter(k -> !projectsOnCache.contains(k) )
+                    .collect(Collectors.toList());
+            if (!newProjectKeys.isEmpty())
+                updateNewProjectsIssues(visitor, newProjectKeys);
+        }
+    }
+
+    private void
```
Comment (jhonypg) [inline core/src/main/java/objective/taskboard/issueBuffer/IssueBufferService.java]: Remover quebra de linha.

File: core/src/main/java/objective/taskboard/jira/JiraIssueJqlBuilderService.java [hunk 4]
```diff
@@ -28,6 +28,13 @@ public String projectsJql(CardRepo cardsRepo) {
         return addTimeConstraintIfPresent(cardsRepo.getLastUpdatedDate(), jql);
     }
 
+    public String projectsFromListWithoutTimeConstraint(List<String> projects){
+        String projectKeys = "'" + projects.stream()
```
Comment (jhonypg) [inline core/src/main/java/objective/taskboard/jira/JiraIssueJqlBuilderService.java]: Acho que você pode refatorar o projectsSqlWithoutTimeConstraint para usar a sua função.

**Pull request merged**